File: advent_of_code/d1/run2.cpp
#include <fstream>
#include <iostream>
#include <string>
#include <vector>
using namespace std;

int main() {
  ifstream inputFile("Input2.txt");
  string s;
  int counter      = 0,
      numIncreases = 0,
      prevSum      = 0,
      currentSum   = 0,
      measurement  = 0;

  vector<int> sliding_window {0,0,0};
  for (int i = 0; i < 3; ++i){
    inputFile >> s;
    measurement = stoi(s);
    sliding_window[i] = measurement;
    prevSum += measurement;
  }
  currentSum = prevSum;

  while (inputFile >> s){
    measurement = stoi(s);

    currentSum = currentSum - sliding_window[counter] + measurement;
    if (currentSum > prevSum)
      ++numIncreases;

    prevSum = currentSum;
    sliding_window[counter] = measurement;
    
    counter = (counter + 1) % 3;
  }

  cout << numIncreases << endl;
  return 0;
}

File: advent_of_code/d1/run2.py
_in = open("input2.txt", 'r').read().split("\n")
N = len(_in)
if N >= 1:
    num_inc = 0
    prevSum = 0
    for i in range(0,len(_in), 3):
        if not _in[i].isnumeric(): continue
        if not _in[i+1].isnumeric(): continue
        if not _in[i+2].isnumeric(): continue
        currSum = int(_in[i]) + int(_in[i+1]) + int(_in[i+2])
        if prevSum > 0 and currSum > prevSum:
            num_inc += 1
        prevSum = currSum
    print(num_inc)
